File: Competitive Programming 3/Section 1.4/Games (Others), Easier/10530 - Guessing Game/Java/Main.java
import java.util.Scanner;
import java.util.ArrayList;

public class Main{
    static Scanner s = new Scanner(System.in);
    static class p{
        int v;
        char flag;
        public p(int v,char flag){
            this.v=v;
            this.flag=flag;
        }
    }  
    public static void main(String[]args){
        while(true){
            ArrayList<p> guesses= new ArrayList<p>();
            boolean honest=true;
            while(true){
                int guessN = Integer.parseInt(s.nextLine());
                if(guessN==0)
                    return;
                char response = s.nextLine().charAt(4);
                if(response=='t'){//output answer
                    for(p el:guesses)
                        if((el.flag=='h'&&el.v<=guessN)||(el.flag=='l'&&el.v>=guessN)){//lie
                            honest=false;
                            break;
                        }
                    break;
                }else{
                    guesses.add(new p(guessN,response));
                }
            }
            System.out.printf("Stan %s\n",honest?"may be honest":"is dishonest");
        }
    }
}

File: Competitive Programming 3/Section 1.4/Games (Others), Easier/10530 - Guessing Game/Python/main.py
def main():
    while True:
        guesses=[]
        while True:
            guessN=int(input())
            if guessN == 0:
                return
            response=input()[4]
            if response=='t':#print answer
                honest=True
                for el in guesses:
                    if (el[1]=='h'and el[0]<=guessN)or(el[1]=='l'and el[0]>=guessN):
                        honest=False
                        break
                print('Stan '+('may be honest' if honest else 'is dishonest'))
                break
            else:
                guesses.append((guessN,response))
main()